### PR TITLE
adding support for log-sink.

### PR DESCRIPTION
**Pull request by jgheewala**

### CODE REVIEW

File: logsink/logsink.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"context"
+)
+
+// Logs is a list of logs
+type Logs []*Log
+
+// Log is a record or event that has occurred
+type Log struct {
+	TraceID        string                 `json:"traceId,omitempty"`
+	SpanID         string                 `json:"spanId,omitempty"`
```
Comment (tigrannajaryan) [inline logsink/logsink.go:13]: `[]byte` probably would be more idiomatic for "byte sequence". `string` probably works too provided that there is no expectation that this should contain valid runes. A comment saying that these are byte sequences is probably sufficient. Do note that not all byte sequences are representable as JSON strings so JSON serialization/deserialization needs to be custom (and data model does not define this).

File: logsink/logsink.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	SeverityText   string                 `json:"severityText,omitempty"`
+	ShortName      string                 `json:"shortName,omitempty"`
+	Body           interface{}            `json:"body"`
+	TimeStamp      int64                  `json:"timestamp"`
```
Comment (tigrannajaryan) [inline logsink/logsink.go:17]: Nit: data model says uint64.

File: logsink/logsink.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	Resource       map[string]interface{} `json:"resource,omitempty"`
+	Attributes     map[string]interface{} `json:"attributes,omitempty"`
+	TraceFlag      byte                   `json:"traceFlags,omitempty"`
+	SeverityNumber int8                   `json:"severityNumber,omitempty"`
```
Comment (tigrannajaryan) [inline logsink/logsink.go:21]: I assume these are at the end for better packing due to field alignment.

File: logsink/logsink.go (hunk 1 above)
Comment (jgheewala) [reply logsink/logsink.go:13]: any example that you have in mind for byte sequences which cannot be represented as JSON strings?

File: logsink/logsink.go (hunk 2 above)
Comment (jgheewala) [reply logsink/logsink.go:17]: ack

File: logsink/logsink.go (hunk 3 above)
Comment (jgheewala) [reply logsink/logsink.go:21]: yes

File: logsink/logsink.go (hunk 1 above)
Comment (jgheewala) [reply logsink/logsink.go]: added comment

File: logsink/log_example.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package logsink
+
+// OTELJsonFormat is useful for testing. All the examples are taken from
+// https://github.com/tigrannajaryan/rfcs/blob/feature/tigran/log-data-model/text/0097-log-data-model.md#example-log-records
```
[the remark this answers is not in the document]
Comment (jgheewala) [reply logsink/log_example.go:4]: it is ok to point to that

File: logsink/logsink.go (hunk 1 above)
Comment (tigrannajaryan) [reply logsink/logsink.go]: JSON is a Unicode text, often UTF-8. Not all byte sequences are valid Unicode sequences. See e.g. https://en.wikipedia.org/wiki/UTF-8#Invalid_byte_sequences
Invalid byte sequences may trigger errors or be silently skipped, etc, depending on JSON implementation.

**Pull request merged**